libeos-updater-util/avahi-service-file.c

```diff
@@ -257,3 +326,426 @@ eos_avahi_service_file_delete (const gchar   *avahi_service_directory,
 
   return TRUE;
 }
+
+static GFile *
+get_ostree_service_file (const gchar *avahi_service_directory,
+                         guint16      repository_index)
+{
+  g_autofree gchar *filename = NULL;
+  g_autofree gchar *service_file_path = NULL;
+
+  filename = g_strdup_printf ("eos-ostree-updater-%" G_GUINT16_FORMAT ".service",
+                              repository_index);
+  service_file_path = g_build_filename (avahi_service_directory,
+                                        filename,
+                                        NULL);
+  return g_file_new_for_path (service_file_path);
+}
+
+static gboolean
+get_summary_timestamp (OstreeRepo *repo,
+                       GCancellable *cancellable,
+                       guint64 *out_timestamp,
+                       GError **error)
+{
+  // TODO: Find the summary without using eos specific bits to ease
+  // the moving of this code to ostree.
+  g_autoptr(EosExtensions) extensions = eos_extensions_new_from_repo (repo,
+                                                                      cancellable,
+                                                                      error);
+
+  if (extensions == NULL)
+    return FALSE;
+
+  // TODO: Try getting the timestamp from summary metadata first.
+  g_assert (out_timestamp);
+  *out_timestamp = extensions->summary_modification_time_unix;
+  return TRUE;
+}
+
+static guint16
+get_repository_index (GVariantDict *options_dict)
+{
+  guint16 index = 0;
+
+  g_variant_dict_lookup (options_dict, EOS_OSTREE_AVAHI_OPTION_REPO_INDEX_Q,
+                         "q", &index);
+
+  return index;
+}
+
+static guint16
+get_avahi_service_port (GVariantDict *options_dict)
+{
+  // TODO: Should we store the port number in the configuration
+  // instead having it as the compile-time constant? I can't remember
+  // why I opted for this solution (was it lack of the configuration
+  // file?)
+  //
+  // In case when config file doesn't specify the port number and the
+  // number wasn't provided with the options_dict variant, likely bail
+  // out.
+  guint16 port = EOS_AVAHI_PORT;
+
+  g_variant_dict_lookup (options_dict, EOS_OSTREE_AVAHI_OPTION_PORT_Q,
+                         "q", &port);
+
+  return port;
+}
+
+static gboolean
+fill_filter_with_refs (OstreeBloom   *filter,
+                       OstreeRepo    *repo,
+                       GCancellable  *cancellable,
+                       GError       **error)
+{
+  g_autoptr(GHashTable) refs = NULL;
+  GHashTableIter iter;
+  gpointer key;
+
+  if (!ostree_repo_list_refs (repo,
+                              NULL, /* no prefix filtering */
+                              &refs,
+                              cancellable,
+                              error))
+    return FALSE;
+
+  g_hash_table_iter_init (&iter, refs);
+  while (g_hash_table_iter_next (&iter, &key, NULL))
```

Comment:
Would be nice to give `key` its proper type (`const gchar *`) rather than it remaining a mysterious `gpointer`. That means adding a cast to the `g_hash_table_iter_next()` call though.